src/core/egg-herbie.rkt

```diff
@@ -10,23 +10,42 @@
 (provide with-egraph egraph-add-expr egraph-run-rules
          egraph-get-simplest egraph-get-variants
          egraph-get-proof egraph-is-unsound-detected
-         rule->egg-rules expand-rules get-canon-rule-name)
+         rule->egg-rules expand-rules get-canon-rule-name
+         remove-rewrites)
+
+
+(define (flatten-let term environment)
```

Comment:
I’m sure we have similar code elsewhere. I assume you looked? (Expand let’s in sugar.rkt) happy to merge even without merging.

Reply:
The syntax is slightly different, unfortunately.
One thing to note here is that there isn't name collision because of the renaming of variables when we go to egglog. The code in rust generates variables like `v0` and `v1`. Something to be careful of in the future

Reply:
Gotit